src/main/java/htsjdk/samtools/util/SamLocusIterator.java

```diff
@@ -369,19 +428,29 @@ private boolean surpassedAccumulationThreshold() {
     }
 
     /**
+     * Check if cigar start with an insertion, ignoring softclip
+     * @param cigar the cigar element
+     * @return <code>true</code> if it start with an insertion; <code>false</code> otherwise
+     */
+    private static boolean startWithInsertion(final Cigar cigar) {
+        for(final CigarElement element: cigar.getCigarElements()) {
+            switch(element.getOperator()) {
+                case I: return true;
+                case S: continue;
```

Comment:
this should be any operator that doesn't consume refbases (except "I"). I think it would be better to implement this as an if statement. Perhaps something like: 
```
<SNIP>
if( element.getOperator()==CigarOperator.I ) return true;
if ( ! element.getOperator().consumesReferenceBases() )  continue;
break;
</SNIP>
```